internal/provider/resource_gitlab_project.go

```diff
@@ -620,6 +621,35 @@ func resourceGitlabProjectCreate(ctx context.Context, d *schema.ResourceData, me
 		}
 	}
 
+	// Branch protection for a newly created branch is an async action, so use WaitForState to ensure it's protected
+	// before we continue. Note this check should only be required when there is a custom default branch set
+	// See issue 800: https://github.com/gitlabhq/terraform-provider-gitlab/issues/800
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"false"},
+		Target:  []string{"true"},
+		Timeout: 2 * time.Minute, //The async action usually completes very quickly, within seconds. Don't wait too long.
+		Refresh: func() (interface{}, string, error) {
+			branch, response, err := client.Branches.GetBranch(project.ID, project.DefaultBranch, gitlab.WithContext(ctx))
+			if err != nil {
+				if response.StatusCode == 404 {
```

Comment:
`response` may be nil here. Prefer to use the `is404(err)` helper function.

Reply:
There's one other reference using similar syntax as this one in the project resource, I'll go ahead and fix both of them tomorrow so we're consistent!

Reply:
Thanks!